sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/CallAsyncClient.java

```diff
@@ -0,0 +1,513 @@
+            Objects.requireNonNull(callId, "'callId' cannot be null.");
+
+            return this.callClient.deleteCallAsync(callId);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Delete a call.
+     *
+     * @param callId Call id.
+     * @return response for a successful DeleteCall request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Void>> deleteCallWithResponse(String callId) {
+        return deleteCallWithResponse(callId, Context.NONE);
+    }
+
+    Mono<Response<Void>> deleteCallWithResponse(String callId, Context context) {
+        try {
+            Objects.requireNonNull(callId, "'callId' cannot be null.");
+            return withContext(contextValue -> {
+                contextValue = context == null ? contextValue : context;
+                return this.callClient.deleteCallWithResponseAsync(callId, contextValue);
+            });
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Cancel Media Operations.
+     *
+     * @param callId The call leg id.
+     * @param operationContext operationContext
+     * @return the response payload of the cancel media operations.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<CancelAllMediaOperationsResponse> cancelAllMediaOperations(String callId, String operationContext) {
+        try {
+            Objects.requireNonNull(callId, "'callId' cannot be null.");
+
+            CancelAllMediaOperationsRequest request = new CancelAllMediaOperationsRequest();
+            request.setOperationContext(operationContext);
+            return this.callClient
+                .cancelAllMediaOperationsAsync(callId, request)
+                .flatMap(response -> Mono.just(response));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Cancel Media Operations.
+     *
+     * @param callId The call leg id.
+     * @param operationContext operationContext
+     * @return the response payload of the cancel media operations.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<CancelAllMediaOperationsResponse>> cancelAllMediaOperationsWithResponse(String callId,
+                                                                                                 String operationContext) {
+        return cancelAllMediaOperationsWithResponse(callId, operationContext, Context.NONE);
+    }
+
+    Mono<Response<CancelAllMediaOperationsResponse>> cancelAllMediaOperationsWithResponse(String callId,
+                                                                                          String operationContext,
+                                                                                          Context context) {
+        try {
+            Objects.requireNonNull(callId, "'callId' cannot be null.");
```

Comment:
Seeing a few duplicate checks here as well. Please clean up

Reply:
duplicate checks removed and cleaned up